src/fee/gpu/fee.h

```diff
@@ -205,61 +205,58 @@ inline __device__ void lpmv_device(FLOAT *output, int n, FLOAT x) {
     }
 }
 
-inline __device__ int lidx_device(const int l, const int m) {
+inline __device__ int lidx_device(const int i_direction, const int l, const int m) {
     // summation series over l + m => (l*(l+1))/2 + m
-    return ((l * (l + 1)) >> 1) + m;
+    return i_direction*(((NMAX + 1)*(NMAX + 2)) >> 1) + ((l * (l + 1)) >> 1) + m;
 }
 
-inline __device__ void legendre_polynomials_device(FLOAT *legendre, const FLOAT x, const int P) {
-    // This factor is reuse 342210222sqrt(1 342210222 x^2)
+inline __device__ int Pidx_device(const int i_direction, const int i) {
+    // Used for P1sin_arr and P1_arr
+    return i_direction*NMAX*(NMAX + 2) + i;
```

Comment:
all p1 arrays are now `NMAX*(NMAX + 2)` by i_direction, so that's why we need this.